nx-X11/programs/Xserver/composite/compext.c

```diff
@@ -27,18 +27,44 @@
 #endif
 
 #include "compint.h"
+#include "XI.h"
+#include "XIproto.h"
+#include "extinit.h"
+
+#ifndef SERVER_COMPOSITE_MAJOR_VERSION
+#define SERVER_COMPOSITE_MAJOR_VERSION 0
+#endif
+
+#ifndef SERVER_COMPOSITE_MINOR_VERSION
+#define SERVER_COMPOSITE_MINOR_VERSION 4
+#endif
 
 static CARD8	CompositeReqCode;
-int		CompositeClientPrivateIndex;
+
+#ifndef NXAGENT_SERVER
```

Comment:
I am not convinced that using NXAGENT_SERVER here is the best choice. The lack of the privates API is not directly related the NXAGENT_SERVER build mode. It is really to the underlying Xserver.

Maybe should rather define something like NX_XSERVER_LACKS_PRIVATES_API in the Imakefile and use that here. Once we fix that (i.e. add include/privates.h to the Xserver code), then we can undefine NX_XSERVER_LACKS_PRIVATES_API in the Imakefile and everything works out well immediately.

We will always have to keep NXAGENT_SERVER everywhere in the code, so esp. here the NXAGENT_SERVER actually would have a misleading meaning.